tests/Console/ConfigurationResolverTest.php

```diff
@@ -868,6 +869,123 @@ public function testResolveRulesWithOption()
         );
     }
 
+    /**
+     * @param string[] $rules
+     * @param string[] $negativeRules
+     *
+     * @dataProvider provideSubtractionCases
+     */
+    public function testResolveRulesWithNegativeSet(array $rules, array $negativeRules)
```

Comment:
I'll have a look later, IIRC I made the test like this to prevent a lot of repeating code and being depended too much on what rules are in the current sets, as it would make the tests fragile.